### ISSUE TITLE
Gumbel trick error

### ISSUE THREAD

**Issue by maestrojeong** [vithursant/VAE-Gumbel-Softmax#1]
https://github.com/vithursant/VAE-Gumbel-Softmax/blob/master/vae_gumbel_softmax.py#L55
You use the logits, however we have to use log(logits) to perform gumbel max trick as stated in [https://timvieira.github.io/blog/post/2014/07/31/gumbel-max-trick/](https://timvieira.github.io/blog/post/2014/07/31/gumbel-max-trick/)

**Comment by j314erre** [comment]
gumbel softmax((log(pi)+g)/tau), where pi are class probabilities, therefore softmax((logits + g)/tau) will suffice.
https://arxiv.org/pdf/1611.01144
